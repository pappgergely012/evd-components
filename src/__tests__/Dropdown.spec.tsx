import React from "react";
import { Dropdown } from "../index";
import renderer, { act } from "react-test-renderer";

interface IData {
  name: string;
  key: number;
}

const data: IData[] = [{name: 'Kis Pizza', key: 1}, {name: 'Nagy Pizza', key: 2}];

const component = renderer.create(
  <Dropdown
    placeholder="Pizzák"
    data={data}
    onItemClicked={(item) => console.log(item)}
    keyExtractor={(item): number => item.key}
    labelExtractor={(item): string => item.name}
  />
);

test("Dropdown list", () => {
  const instance   = component.root;
  const inst: any  = component.toJSON();

  expect(instance.props.placeholder).toBe("Pizzák");
  expect(instance.props.data).toBeInstanceOf(Array);

  act(() => {
    inst.children[0].props.onClick();
  })

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Dropdown list selects the first element", () => {
  const instance   = component.root;

  expect(instance.props.placeholder).toBe("Pizzák");
  expect(instance.props.data).toBeInstanceOf(Array);

  act(() => {
    //the list test opened the dropdown
    const openedInstance: any = component.toJSON();

    openedInstance.children[1].children[0].props.onClick(data[0]);
  });

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});