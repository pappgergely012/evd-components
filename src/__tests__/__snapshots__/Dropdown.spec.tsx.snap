// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown list 1`] = `
<div
  className="dropdown open"
>
  <div
    className="dropdown-row"
    onClick={[Function]}
  >
    <span
      className="dropdown-title"
    >
      Pizz√°k
    </span>
    <span
      className="dropdown-arrow"
    />
  </div>
  <div
    className="dropdown-list"
  >
    <div
      className="dropdown-item"
      onClick={[Function]}
    >
      Kis Pizza
    </div>
    <div
      className="dropdown-item"
      onClick={[Function]}
    >
      Nagy Pizza
    </div>
  </div>
</div>
`;

exports[`Dropdown list selects the first element 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown-row"
    onClick={[Function]}
  >
    <span
      className="dropdown-title"
    >
      Kis Pizza
    </span>
    <span
      className="dropdown-arrow"
    />
  </div>
</div>
`;
